api/src/pcapi/routes/backoffice/bookings/individual_bookings_blueprint.py

```diff
@@ -95,10 +96,11 @@ def _get_individual_bookings(
     if form.q.data:
         search_query = form.q.data
 
-        if BOOKING_TOKEN_RE.match(search_query):
-            or_filters.append(bookings_models.Booking.token == search_query.upper())
+        terms = search_utils.split_terms(search_query)
+        if all(BOOKING_TOKEN_RE.match(term) for term in terms):
+            or_filters += [bookings_models.Booking.token == term.upper() for term in terms]
```

Comment:
Je ne suis pas sur de comprendre pourquoi on ne fait la recherche par contremarque que seulement si tous les termes du champ valide le format ?
Pourquoi ne pas juste

```suggestion
        for term in terms:
            if BOOKING_TOKEN_RE.match(term):
                or_filters += [bookings_models.Booking.token == term.upper()]
```

Reply:
On cherche ici à détecter si le champ de ce ticket contient une liste de contremarques, afin de chercher uniquement dans la colonne `Booking.token`, et ne pas perdre de temps sur une requête dans le nom du bénéficiaire, bien plus longue à répondre.

Par exemple, si je cherche « damien ramelet » dans le champ de recherche : 
- avec ma solution, je cherche dans le nom car « ramelet » ne _match_ pas un format de code contremarque ;
- avec ta proposition, je cherche le token « damien » dans les contremarques, et comme il y a des filtres détectés dans `or_filters`, alors je ne cherche rien dans le nom